frontend/benefit/applicant/src/components/applications/forms/application/step3/attachmentsList/AttachmentsList.tsx

```diff
@@ -43,9 +69,23 @@ const AttachmentsList: React.FC<AttachmentsListProps> = ({
           )}
         </>
       )}
-      <$PrimaryButton style={{ width: 'auto' }} iconLeft={<IconPlus />}>
-        {t(`${translationsBase}.add`)}
-      </$PrimaryButton>
+      <$UploadContainer onClick={handleUploadClick}>
+        <$PrimaryButton
+          isLoading={isUploading}
+          loadingText={t(`common:upload.isUploading`)}
+          style={{ width: 'auto' }}
+          iconLeft={<IconPlus />}
+        >
+          {t(`${translationsBase}.add`)}
+        </$PrimaryButton>
+        <input
```

Comment:
So you decided to not use the HDS component, or we will replace this later?

Reply:
its not implemented yet:)

Reply:
and i checked it, its not what we need actually.... but lets see if we can reuse it at some point